ts/components/ui/ButtonSolid.tsx

```diff
@@ -0,0 +1,218 @@
+import * as React from "react";
+import { useCallback } from "react";
+import {
+  StyleSheet,
+  Pressable,
+  GestureResponderEvent
+  // PixelRatio
+} from "react-native";
+import Animated, {
+  useSharedValue,
+  useAnimatedStyle,
+  withSpring,
+  useDerivedValue,
+  interpolate,
+  Extrapolate,
+  interpolateColor
+} from "react-native-reanimated";
+import themeVariables from "../../theme/variables";
+import { IOColors, IOColorType } from "../core/variables/IOColors";
+import { IOSpringValues, IOScaleValues } from "../core/variables/IOAnimations";
+import { BaseTypography } from "../core/typography/BaseTypography";
+import { IOButtonStyles } from "../core/variables/IOStyles";
+import { WithTestID } from "../../types/WithTestID";
+
+type Props = WithTestID<{
+  color?: "primary" | "danger" | "contrast";
+  label: string;
+  small?: boolean;
+  fullWidth?: boolean;
+  disabled?: boolean;
+  accessibilityLabel: string;
+  accessibilityHint?: string;
+  onPress: (event: GestureResponderEvent) => void;
+}>;
+
+type ColorStates = {
+  default: string;
+  pressed: string;
+  label: {
+    default: IOColorType;
+    disabled: IOColorType;
+  };
+};
+
+// COMPONENT CONFIGURATION
+
+const mapColorStates: Record<NonNullable<Props["color"]>, ColorStates> = {
+  // Primary button
+  primary: {
+    default: IOColors.blue,
+    pressed: IOColors.blueUltraLight,
+    label: {
+      default: "white",
+      disabled: "white"
+    }
+  },
+  // Danger button
+  danger: {
+    default: IOColors.red,
+    pressed: IOColors.red,
```

Comment:
I suppose that `danger` buttons looks the same in the pressed state, are there any UX reasons for this? Wouldn't a slightly lighter (or darker) red be better?

Reply:
Exactly! Because some states needed the new color palette, which was not yet available when the PR was opened. According to the design (not yet visible), all states will have a different color for the pressed state.